### PR TITLE
began implementation of light sensor controller and helper

### PR DESCRIPTION
**Pull request by jyanev**
added constructor for light sensor controller and helper

### CODE REVIEW

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    private long startTime;
+    private long endTime;
+
+    private int darknessThreshold;
+    private int darknessValue;
+
+
+    public LightSensorHelper(LEDHelper ledHelper,
+                             @Named("photo-resistor-input") DigitalInput sensorInput,
```
Comment (SwindleA) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java:33]: In other components, the @Named things are set in the Controller.

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                             @Named("photo-resistor-output") DigitalOutput sensorOutput) {
+        this.ledHelper = ledHelper;
```
Comment (SwindleA) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java:35]: The helper should only interact with the light sensor.  Interactions with the led should be taken care of in the controller.

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        this.sensorInput = sensorInput;
+        this.sensorOutput = sensorOutput;
+
+        initializePhotoSensor();
+    }
+
+    private void initializePhotoSensor() {
```
Comment (SwindleA) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java:42]: I think this should be moved to the controller.

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java (hunk 1 above)
Comment (SwindleA) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java:33]: In other components, the `@Named` things are set in the Controller.

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java (hunk 2 above)
Comment (SwindleA) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java:35]: The helper should only interact with the light sensor. Interactions with the led should be taken care of in the controller.

**Review by SwindleA — CHANGES_REQUESTED** [review]
There are several things that should be moved from the helper to the controller.  I think it should be discussed what should be done with `Darkness.java`. Do we try and move the darkness code into the helper, to minimize the number of files, or do we establishes how to store the helper files to a helper?

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/Darkness.java
Comment (ruthvikm) [inline micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/Darkness.java:1]: We use only one helper class to provide support for a component. Please make necessary corrections and merge the content of this file to LightSensorHelper.java.

File: micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java (hunk 1 above)
Comment (ruthvikm) [reply micronautpi4j-utils/src/main/java/com/opensourcewithslu/inputdevices/LightSensorHelper.java:33]: I agree with Adrian

**Review by ruthvikm — CHANGES_REQUESTED** [review]
Test your code on raspberry pi and share the results. If it is working, then we can merge this component's code to main

**Comment by ruthvikm** [comment]
working on a new branch

**Pull request closed without merging**